### PR TITLE
Added support for btl trace files to GC Benchmarking Infra

### PR DESCRIPTION
**Pull request by ivdiazsa**
Btl is another file format for traces that is fully supported by PerfView. GC Benchmarking Infra did not recognize it and therefore failed. This PR adds the necessary support to be able to analyze these traces as well.

### CODE REVIEW

**Comment by ivdiazsa** [comment]
@Maoni0 @VSadov @cshung

File: src/benchmarks/gc/src/analysis/managed-lib/Analysis.cs [hunk 1]
```diff
@@ -505,18 +505,20 @@ public class TracedProcesses
 
         public static void PrintEventsWithoutTraceLog(string tracePath, Func<TraceEvent, bool> filter, uint? maxEvents)
         {
-            using TraceEventDispatcher source = getSource(tracePath);
+            using TraceEventDispatcher source = TraceEventDispatcher.GetDispatcherFromFileName(tracePath);
```
Comment (Maoni0) [inline src/benchmarks/gc/src/analysis/managed-lib/Analysis.cs]: >using TraceEventDispatcher source = TraceEventDispatcher.GetDispatcherFromFileName(tracePath); [](start = 12, length = 94)

you could do
`using TraceEventDispatcher source = TraceEventDispatcher.GetDispatcherFromFileName(tracePath)!;`

that way you don't need to do source! at all the other places.

**Comment by Maoni0** [comment]
otherwise LGTM! also CC @brianrob

**Comment by brianrob** [comment]
Awesome.  LGTM as well.

**Comment by ivdiazsa** [comment]
Thanks! I had to leave the '!' in a few statements because, in spite of adding '!' to the declaration of `source`, dotnet still complained about those cases, giving an error saying `Dereference of possible null reference`.

**Comment by brianrob** [comment]
@ivdiazsa, @Maoni0 let me know when you're happy with this and I will merge it.

**Comment by ivdiazsa** [comment]
Thanks Brian! I see that all the checks have passed, so it's now ready for merging. @brianrob

**Pull request merged**